src/client/game.cpp

```diff
@@ -3829,13 +3826,26 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats, f32 dtime,
 	   1. Delete formspec menu reference if menu was removed
 	   2. Else, make sure formspec menu is on top
 	*/
-	if (current_formspec) {
-		if (current_formspec->getReferenceCount() == 1) {
-			current_formspec->drop();
-			current_formspec = NULL;
-		} else if (isMenuActive()) {
-			guiroot->bringToFront(current_formspec);
+	auto formspec = m_game_ui->getFormspecGUI();
+	while (formspec) { // breakable. only runs for one iteration
```

Comment:
I believe this is common notation:

```c++
for (;;) {
```

or maybe this is better:

```c++
do {


} while (false);
```

I'm not a fan of this way of coding though

Reply:
Why not a plain old `if`?

Reply:
@rubenwardy I'll use the second suggestion, thanks.
@ClobberXD Because the flow of this code (abort if X or Y is true) makes `if`s very hard to use.

Reply:
Use `goto`! 8D